### PR TITLE
Fix loading timezone info on windows

### PR DESCRIPTION
**Pull request by undef1nd**
**What this PR does / why we need it**:

`date_format`.`default_timezone` configuration setting doesn't work on Windows without Go installed.
When it is set to a timezone value, there’s a message in a log `"Unknown timezone as a default_timezone"`

It happens due to `time.LoadLocation` returning an error: when called it scans the dir/file named by the `ZONEINFO` environment variable. Though we bundle `zoneinfo.zip` into Grafana package, we don’t set `ZONEINFO` environment variable with a path to `tools\zoneinfo.zip` file.

This issue impacts the Enterprise reporting feature: once `time.LoadLocation` is called, the env variable is not being evaluated again (even if it’s set in the reporting service Init).

**Special notes for your reviewer** 

It may be not the best solution. If in the future we add `time.LoadLocation` call that happens before the environment variable is set, it will reintroduce the issue. Is there a better place in the code where I could put this (given that it depends on the HomePath)?

### CODE REVIEW

**Review by aknuds1 — COMMENTED** [review]
Might it be better to use a build tag to compile Windows specific code only on Windows, instead of using runtime OS detection?

**Comment by aknuds1** [comment]
Maybe setting the zone info could be done on Grafana initialization (i.e. after `HomePath` is configured)?

**Comment by undef1nd** [comment]
> Maybe setting the zone info could be done on Grafana initialization (i.e. after `HomePath` is configured)?

Somewhere [here](https://github.com/grafana/grafana/blob/master/pkg/server/server.go#L282) or maybe still better [here](https://github.com/grafana/grafana/blob/master/pkg/setting/setting.go#L759) ?

**Comment by aknuds1** [comment]
> > Maybe setting the zone info could be done on Grafana initialization (i.e. after `HomePath` is configured)?
> 
> Somewhere [here](https://github.com/grafana/grafana/blob/master/pkg/server/server.go#L282) or maybe still better [here](https://github.com/grafana/grafana/blob/master/pkg/setting/setting.go#L759) ?

I think the latter location looks the best, unless I'm overlooking something.

File: pkg/setting/setting.go [hunk 1]
```diff
@@ -757,6 +761,14 @@ func (cfg *Cfg) validateStaticRootPath() error {
 func (cfg *Cfg) Load(args *CommandLineArgs) error {
 	setHomePath(args)
 
+	// Fix for missing IANA db on Windows
+	_, zoneInfoSet := os.LookupEnv(zoneInfo)
+	if runtime.GOOS == "windows" && !zoneInfoSet {
+		if err := os.Setenv(zoneInfo, HomePath+"/tools/zoneinfo.zip"); err != nil {
```
Comment (aknuds1) [inline pkg/setting/setting.go:767]: BTW, shouldn't you instead use the following to get the right path separator on Windows?

```suggestion
		if err := os.Setenv(zoneInfo, filepath.Join(HomePath, "tools", "zoneinfo.zip")); err != nil {
```

Comment (undef1nd) [reply pkg/setting/setting.go:767]: yes, probably I should. Thanks you. Though somehow the concatenated strings version worked on Win as well 🤔

**Review by aknuds1 — APPROVED** [review]
LGTM

**Review by xlson — APPROVED** [review]

**Pull request merged**